opentelemetry-kotlin-implementation/src/commonMain/kotlin/io/embrace/opentelemetry/kotlin/tracing/model/SpanImpl.kt

```diff
@@ -0,0 +1,129 @@
+package io.embrace.opentelemetry.kotlin.tracing.model
+
+import io.embrace.opentelemetry.kotlin.ExperimentalApi
+import io.embrace.opentelemetry.kotlin.attributes.MutableAttributeContainer
+import io.embrace.opentelemetry.kotlin.attributes.MutableAttributeContainerImpl
+import io.embrace.opentelemetry.kotlin.tracing.data.EventData
+import io.embrace.opentelemetry.kotlin.tracing.data.LinkData
+import io.embrace.opentelemetry.kotlin.tracing.data.StatusData
+
+@OptIn(ExperimentalApi::class)
+internal class SpanImpl(
+    private val attrs: MutableAttributeContainer = MutableAttributeContainerImpl()
+) : Span {
+
+    // in future this may need implementing as a tri-state enum to better support span processors
+    private var recording = true
+
+    override var name: String = ""
+        get() = throw UnsupportedOperationException()
+
+    override var status: StatusData = StatusData.Unset
+        get() = throw UnsupportedOperationException()
+
+    override fun end() {
+        recording = false
+    }
+
+    override fun end(timestamp: Long) {
+        throw UnsupportedOperationException()
+    }
+
+    override fun isRecording(): Boolean = recording
```

Comment:
I'm not 100% on what our approach to synchronization should be around ending a span, so figured this is a good candidate for opening it up to discussion. Do we want to use a lock to check whether the span is ended like the Java SDK does & risk slow perf? Or should we take a copy of the span object when it ends & ignore any subsequent updates, even though they might appear in this referenced object?